tensorflow/core/grappler/optimizers/remapper.cc

```diff
@@ -1881,15 +1898,14 @@ Status Remapper::Optimize(Cluster* cluster, const GrapplerItem& item,
     // Remap Conv2D+FusedBatchNorm+Activation into the _FusedConv2D;
     ContractionWithBatchNormAndActivation
         contract_with_batch_norm_and_activation;
-    if (allow_non_differentiable_rewrites &&
+    if (!is_onednn_enabled && allow_non_differentiable_rewrites &&
```

Comment:
Same question here.
```suggestion
    if (is_onednn_enabled && allow_non_differentiable_rewrites &&
```

Reply:
We do not support contraction with fused batch norm for now.  We will have a PR (https://github.com/tensorflow/tensorflow/pull/47828) to support this case. 

So the code is logically correct.